metricproviders/kayenta/kayenta.go

```diff
@@ -0,0 +1,202 @@
+package kayenta
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	log "github.com/sirupsen/logrus"
+	"io/ioutil"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/tidwall/gjson"
+
+	"github.com/argoproj/argo-rollouts/pkg/apis/rollouts/v1alpha1"
+
+	metricutil "github.com/argoproj/argo-rollouts/utils/metric"
+	templateutil "github.com/argoproj/argo-rollouts/utils/template"
+)
+
+const (
+	//ProviderType indicates the provider is kayenta
+	ProviderType = "Kayenta"
+	KayentaScoreURL string = "{{inputs.address}}/canary/{{inputs.canaryExecutionId}}"
+
+	JobURL string = "{{inputs.address}}/canary/{{inputs.canaryConfigId}}?application={{inputs.application}}&metricsAccountName={{inputs.metricsAccountName}}&configurationAccountName={{inputs.configurationAccountName}}&storageAccountName={{inputs.storageAccountName}}"
+
+	JobPayloadTemplate string = `
+							{
+								"scopes": {
+										{{inputs.scopes}}
+								},
+                                "thresholds" : {
+                                    "pass": {{inputs.pass}},
+                                    "marginal": {{inputs.marginal}}
+                                }
+                            }`
+
+
+)
+
+type Provider struct {
+	logCtx log.Entry
+	client http.Client
+}
+
+// Type incidates provider is a kayenta provider
+func (p *Provider) Type() string {
+	return ProviderType
+}
+
+// Run queries kayentd for the metric
+func (p *Provider) Run(run *v1alpha1.AnalysisRun, metric v1alpha1.Metric) v1alpha1.Measurement {
+	startTime := metav1.Now()
+	newMeasurement := v1alpha1.Measurement{
+		StartedAt: &startTime,
+	}
+
+	jobURL := strings.Replace(JobURL, "{{inputs.address}}", metric.Provider.Kayenta.Address, 1)
+	jobURL = strings.Replace(jobURL, "{{inputs.canaryConfigId}}", metric.Provider.Kayenta.CanaryConfigId, 1)
```

Comment:
My idea is to support both name and id available and I can add the support for name lookup later.  Since id is an immutable unique identifier, I don't think it is a good idea to have to make the roundtrip every time to look up the id from name.